### PR TITLE
EVG-12955 edit commit message

### PR DESCRIPTION
**Pull request by ybrill**
When a user enqueues a patch we've been using the patch description as the commit message. Let users edit the commit message at enqueue time.

Some thoughts/misgivings: 
-  design incorporated the commit message into the existing popconfirm, but when I implemented it a modal seemed more intuitive. I've implemented it as a modal. Do I need to follow up with design?
- How do I deploy this change? I needed to change the enqueuePatch mutation to include a new commit message parameter. I think the mutation won't work when only one side of the interface is on the new version. Do I add a second duplicate mutation and remove the old one once they're both deployed? Is there another way?

### CODE REVIEW

**Comment by ybrill** [comment]
Goes together with https://github.com/evergreen-ci/evergreen/pull/4283

**Comment by khelif96** [comment]
Can you merge master in?

**Comment by ybrill** [comment]
evergreen retry

**Comment by ybrill** [comment]
@khelif96 I merged master. 
I think the remaining issues are because I haven't yet merged the schema changes on the backend.

File: src/components/PatchActionButtons/EnqueuePatch.tsx [hunk 1]
```diff
@@ -1,78 +1,54 @@
...
-import { usePatchAnalytics } from "analytics";
 import { DropdownItem } from "components/ButtonDropdown";
-import { useBannerDispatchContext } from "context/banners";
-import {
-  EnqueuePatchMutation,
-  EnqueuePatchMutationVariables,
-} from "gql/generated/types";
-import { ENQUEUE_PATCH } from "gql/mutations";
-import { StyledBody } from "./UnschedulePatchTasks";
+import { EnqueuePatchModal } from "pages/patch/index";
```
Comment (khelif96) [inline src/components/PatchActionButtons/EnqueuePatch.tsx:4]: You can just do `import { EnqueuePatchModal } from "pages/patch";`

File: src/pages/patch/EnqueueModal.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+              patchAnalytics.sendEvent({ name: "Enqueue" });
+            }}
+            variant="primary"
+          >
+            Enqueue
+          </Button>,
+        ]}
+        data-cy="enqueue-modal"
+      >
+        <>
```
Comment (khelif96) [inline src/pages/patch/EnqueueModal.tsx:87]: This fragment can probably be removed.

**Review by khelif96 — APPROVED** [review]

File: src/pages/patch/EnqueueModal.tsx (hunk 2 above)
Comment (ybrill) [reply src/pages/patch/EnqueueModal.tsx]: done

File: src/components/PatchActionButtons/EnqueuePatch.tsx (hunk 1 above)
Comment (ybrill) [reply src/components/PatchActionButtons/EnqueuePatch.tsx:4]: When I tried this it thought I was trying to import from `pages/Patch.tsx`

**Comment by khelif96** [comment]
Tests should pass once the backend portion of this is merged. After it is you should try to merge in master and run `yarn codegen` one more time.

**Review by SupaJoon — APPROVED** [review]

**Pull request merged**